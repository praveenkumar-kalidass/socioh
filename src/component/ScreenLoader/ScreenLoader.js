import React from 'react';
import { Modal } from 'react-native';
import PropTypes from 'prop-types';
import { Spinner } from 'native-base';

import { COLOR } from '../../constant';
import { OverlayContainer } from './ScreenLoader.style';

const ScreenLoader = ({ isVisible }) => (
  <Modal visible={isVisible} testID="screen_loader_modal" transparent>
    <OverlayContainer>
      <Spinner testID="screen_loader" color={COLOR.PRIMARY} />
    </OverlayContainer>
  </Modal>
);

ScreenLoader.defaultProps = {
  isVisible: false,
};

ScreenLoader.propTypes = {
  isVisible: PropTypes.bool,
};

export default ScreenLoader;
