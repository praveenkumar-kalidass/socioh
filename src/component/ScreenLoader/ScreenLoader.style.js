import styled from 'styled-components/native';

export const OverlayContainer = styled.View`
  align-items: center;
  background-color: rgba(255, 255, 255, 0.5);
  flex: 1;
  justify-content: center;
`;
