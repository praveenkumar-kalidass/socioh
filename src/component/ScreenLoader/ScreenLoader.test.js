import React from 'react';
import { render } from '@testing-library/react-native';

import ScreenLoader from './index';

describe('ScreenLoader', () => {
  it('should match snapshot', () => {
    const container = render(<ScreenLoader />);

    expect(container).toMatchSnapshot();
  });

  it('should render loader when isVisible prop is true', () => {
    const { getByTestId } = render(<ScreenLoader isVisible />);

    expect(getByTestId('screen_loader_modal').props.visible).toStrictEqual(
      true,
    );
  });

  it('should not render loader when isVisible prop is undefined or false', () => {
    const { getByTestId } = render(<ScreenLoader />);

    expect(getByTestId('screen_loader_modal').props.visible).toStrictEqual(
      false,
    );
  });
});
