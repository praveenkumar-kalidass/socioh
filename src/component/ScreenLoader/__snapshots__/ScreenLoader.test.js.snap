// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenLoader should match snapshot 1`] = `
<Modal
  hardwareAccelerated={false}
  testID="screen_loader_modal"
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#1abfbf"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "height": 80,
        }
      }
      testID="screen_loader"
    />
  </View>
</Modal>
`;
